### DISCUSSION TITLE
How do I revert / discard specific lines of a file when staging prior to a commit?

### DISCUSSION THREAD

**Discussion by Moult** [jesseduffield/lazygit#3460] [Q&A] [answered]
If I select a file and press `<enter>`, then I can see hunks. I can either line by line press space to toggle that line as staged or unstaged, or use `<v>` to visual select then stage my selection. However, how do I revert the visual selection or a single line?

In git gui, I can select line(s) then right click and press revert lines/hunk.

I saw this discussion https://github.com/jesseduffield/lazygit/discussions/3157 but that seems to be after you commit, whereas I'm talking about prior to committing, when I'm busy staging.

Awesome tool by the way I love it!

**Comment by mark2185** [comment]
Try pressing `<Enter>` on a file to get the staged and unstaged changes view, and then press `<Tab>` to switch between the two.

**Reply by stefanhaller** [reply, marked as answer]
I don't think that was the question.

What git gui calls "revert" is called "discard" in lazygit. (That's a better name because "revert" has a different meaning in git.)

To do that, select the lines that you want to discard (either using `v`, or just by pressing shift-down-arrow), and hit `d` for "discard". `d` is a very general keybinding in lazygit for discarding, deleting, or dropping all sorts of things.

**Reply by Moult** [reply]
Thanks so much that was what I was after! (I vaguely recall trying it out but not getting the intended results, I must've messed up somehow)
